xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-page/src/main/java/org/xwiki/test/page/LocalizationSetup.java

```diff
@@ -52,15 +55,19 @@ public static void setUp(TestComponentManager tcm) throws Exception
     {
         LocalizationScriptService lss = mock(LocalizationScriptService.class);
         tcm.registerComponent(ScriptService.class, "localization", lss);
-        when(lss.render(anyString())).thenAnswer(
-            new Answer<String>() {
-                @Override
-                public String answer(InvocationOnMock invocationOnMock) throws Throwable
-                {
-                    // Return the translation key as the value
-                    return invocationOnMock.getArgument(0);
-                }
-            }
-        );
+        when(lss.render(anyString())).thenAnswer(invocationOnMock -> {
+            // Return the translation key as the value
+            return invocationOnMock.getArgument(0);
+        });
+        when(lss.render(anyString(), anyCollection())).thenAnswer((Answer<String>) invocationOnMock -> {
+            // Displays the comma separated list of parameters between squared brackets after the translation key as 
+            // the value.
+            // For instance: my.key [paramA, paramB]
+            Object key = invocationOnMock.getArgument(0);
+            Collection<?> parameters = invocationOnMock.getArgument(1);
+            return parameters.stream()
+                .map(String::valueOf)
+                .collect(Collectors.joining(", ", key + " [", "]"));
+        });
```

Comment:
I think we need to review the mocking we do in PageTest since we talked about mocking the minimum now.